app/controllers/v0/sessions_controller.rb

```diff
@@ -180,17 +182,29 @@ def build_logout_errors(logout_response, logout_request, session, user)
       errors
     end
 
-    def saml_callback_success_url
+    def default_relay_url
+      Settings.saml.relays.vetsgov
+    end
+
+    def saml_login_relay_url
+      return default_relay_url if current_user.nil?
+      # TODO: this validation should happen when we create the user, not here
+      if current_user.loa.key?(:highest) == false || current_user.loa[:highest].nil?
+        log_message_to_sentry('ID.me did not provide LOA.highest!', :error)
+        return default_relay_url
+      end
+
       if current_user.loa[:current] < current_user.loa[:highest]
-        SAML::SettingsService.idme_loa3_url(current_user)
+        SAML::SettingsService.idme_loa3_url(current_user, success_relay: params['RelayState'])
+      elsif valid_relay_state?
+        params['RelayState']
```

Comment:
@omgitsbillryan I'm reading this as opening up the redirect to non-whitelisted external input. Is there protection I'm not seeing?